### DISCUSSION TITLE
Unknown Version?

### DISCUSSION THREAD

**Discussion by geoffreywiseman** [vengroff/usingversion#6] [Q&A] [answered]
Hi @vengroff (or anyone else listening):

I was updating [awswl](https://pypi.org/project/awswl/), and noticed that the method I'd used to single-source the module version was showing the installed package version rather than what I was working on, and after digging around in the poetry issue for single-sourced versions, came across this package, which worked well in the development cycle.

After publishing to pypi, I discovered that the cli is now reporting `unknown` version:
```
❯ awswl version
awswl vunknown
```

Obviously not as desired -- is it something about how I integrated it? Weird edge case? What am I missing?  Haven't attempted to debug an issue only present in the installed version before. ;)

**Comment by vengroff** [comment]
I'd be happy to take a look. Can you paste in the code or point me to the code that gets invoked when you run from the command line like that? Then I can try to see what the issue might be.

**Reply by geoffreywiseman** [reply]
Sure -- and I know this isn't a support chat, I'm just a little puzzled by the change in how it behaved in development and how it's behaving post-install.  I may well be be able to figure out how to debug the installed version, but I was kinda hoping this might be something you'd already encountered (sounds like not, though).

Anyway:
- The normal entry point is through [the script that poetry makes](https://github.com/geoffreywiseman/awswl/blob/2c628abdceb3113a8c6c418a5e36801512dfcbab/pyproject.toml#L28).
- That invokes this [main](https://github.com/geoffreywiseman/awswl/blob/2c628abdceb3113a8c6c418a5e36801512dfcbab/awswl/main.py#L7) method.
- There's some argparse stuff in `cli.py`, but shouldn't really impact this. Once the parsing's done, the [execute](https://github.com/geoffreywiseman/awswl/blob/2c628abdceb3113a8c6c418a5e36801512dfcbab/awswl/main.py#L15) method will end up invoking [cmd_version](https://github.com/geoffreywiseman/awswl/blob/2c628abdceb3113a8c6c418a5e36801512dfcbab/awswl/commands.py#L160) in `commands.py`, which literally just prints the value from the module.
- `useversion` is bootstrapped as suggested [with `__init__.py`](https://github.com/geoffreywiseman/awswl/blob/master/awswl/__init__.py)

Don't think there are many surprises there, but obviously something isn't really working as planned, and I haven't figured out what yet.

**Reply by geoffreywiseman** [reply, marked as answer]
Aw, crap. I think I just saw the problem.

I noticed in `__init__.py` I had `mypackage` still and that looked ... unlikely.  I've just checked your README again and it definitely says to replace that with my package name.

Interestingly, that didn't pose a problem during the development cycle.

**Reply by geoffreywiseman** [reply]
Yup -- I just edited the file where pip put it `/usr/local/lib/python3.9/site-packages/awswl/__init__.py`, and changed the package name and that fixed it.  Guess I'll make myself a patch version.

Thanks for being the rubber duck for my rubber duck debugging. ;)

**Reply by vengroff** [reply]
Glad to here you figured it out!
